### ISSUE TITLE
How to Understand BERT Model with Multiple [CLS] Output？

### ISSUE THREAD

**Issue by quant-science** [nlpyang/PreSumm#120]
class ExtSummarizer(nn.Module):
    def forward(self, src, segs, clss, mask_src, mask_cls):
        top_vec = self.bert(src, segs, mask_src)
        sents_vec = top_vec[torch.arange(top_vec.size(0)).unsqueeze(1), clss]
        sents_vec = sents_vec * mask_cls[:, :, None].float()
        sent_scores = self.ext_layer(sents_vec, mask_cls).squeeze(-1)
        return sent_scores, mask_cls

i guess the above code extract serveral BERT [CLS] outputs, but in my understanding, BERT model only have one [CLS] output?  
hope someone can help me figure it out

**Comment by daidew** [comment]
The paper said that for extractive summarization to work, it needs multiple CLS tokens to tell the model that there are many labels that we're predicting. So the output of the same position of CLS token is the probability that we should select ith sentence as a part of the summary.
